cmd/collector/app/span_processor_test.go

```diff
@@ -807,3 +821,179 @@ func TestSpanProcessorWithOnDroppedSpanOption(t *testing.T) {
 	require.EqualError(t, err, processor.ErrBusy.Error())
 	assert.Equal(t, []string{"op3"}, droppedOperations)
 }
+
+func optionsWithPorts(portHttp string, portGrpc string) *cFlags.CollectorOptions {
+	opts := &cFlags.CollectorOptions{
+		OTLP: struct {
+			Enabled bool
+			GRPC    configgrpc.ServerConfig
+			HTTP    confighttp.ServerConfig
+		}{
+			Enabled: true,
+			HTTP: confighttp.ServerConfig{
+				Endpoint: portHttp,
+			},
+			GRPC: configgrpc.ServerConfig{
+				NetAddr: confignet.AddrConfig{
+					Endpoint:  portGrpc,
+					Transport: confignet.TransportTypeTCP,
+				},
+			},
+		},
+	}
+	return opts
+}
+
+func TestOTLPReceiverWithV2Storage(t *testing.T) {
+	// Setup mock writer and expectations
+	mockWriter := mocks.NewWriter(t)
+	traces := ptrace.NewTraces()
+	rSpans := traces.ResourceSpans().AppendEmpty()
+	sSpans := rSpans.ScopeSpans().AppendEmpty()
+	span := sSpans.Spans().AppendEmpty()
+	span.SetName("test-trace")
+
+	var receivedTraces atomic.Pointer[ptrace.Traces]
+	var recievedCtx atomic.Pointer[context.Context]
+	mockWriter.On("WriteTraces", mock.Anything, mock.Anything).
+		Run(func(args mock.Arguments) {
+			storeContext := args.Get(0).(context.Context)
+			storeTrace := args.Get(1).(ptrace.Traces)
+			receivedTraces.Store(&storeTrace)
+			recievedCtx.Store(&storeContext)
+		}).Return(nil)
+
+	spanProcessor, err := NewSpanProcessor(
+		mockWriter,
+		nil,
+		Options.NumWorkers(1),
+		Options.QueueSize(1),
+		Options.ReportBusy(true),
+	)
+	require.NoError(t, err)
+	defer spanProcessor.Close()
+	logger := zaptest.NewLogger(t)
+
+	portHttp := "4317"
+	portGrpc := "4318"
+
+	// Send trace via HTTP
+	t.Run("Send trace data using HTTP connection", func(t *testing.T) {
+		// Can't send tenancy headers with http request to OTLP receiver
+		tenancyMgr := &tenancy.Manager{
+			Enabled: false,
+		}
+
+		// Create and start receiver
+		rec, err := handler.StartOTLPReceiver(
+			optionsWithPorts(fmt.Sprintf("localhost:%v", portHttp), fmt.Sprintf("localhost:%v", portGrpc)),
+			logger,
+			spanProcessor,
+			tenancyMgr,
+		)
+		require.NoError(t, err)
+		ctx := context.Background()
+		defer rec.Shutdown(ctx)
+
+		url := fmt.Sprintf("http://localhost:%v/v1/traces", portHttp)
+		client := &http.Client{}
+
+		marshaler := ptrace.JSONMarshaler{}
+		data, err := marshaler.MarshalTraces(traces)
```

Comment:
you are defining traces in the root function, but it's only used in this test. What about just sending an explicit JSON string that can be defined inline?